### PR TITLE
Add specs for !global deprecation warnings

### PR DESCRIPTION
**Pull request by nex3**
See sass/sass#2606

[skip dart-sass]

### CODE REVIEW

**Comment by nshahan** [comment]
Are there any specs for global variables that do not generate a warning?

**Comment by nex3** [comment]
There are a bunch of them already throughout the repo. Eventually I'd like to organize them in one place, but that's out of scope for this change.

**Comment by nshahan** [comment]
OK, this spec just felt a little out of place sitting at the top level. I took a quick glance to see if there was already a section for variables but it didn't jump out at me.

File: spec/libsass-closed-issues/issue_759.hrx [hunk 1]
```diff
@@ -21,3 +28,49 @@ foo {
...
+  | ^^^^^^^^^^^^^^^^^^^^^^^^^
+  '
+    /sass/spec/libsass-issues/issue_759/input.scss 2:1  root stylesheet
+
+DEPRECATION WARNING: As of Dart Sass 2.0.0, !global assignments won't be able to
+declare new variables. Consider adding `$c: null` at the top level.
+
+  ,
+3 | $c: 30px !default !default !default !global !global !global;
+  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (nshahan) [inline spec/libsass-closed-issues/issue_759.hrx]: Is there an issue for allowing `!default` or `!global` at most once? I believe it was mentioned in the written spec you added recently.

Comment (nex3) [reply spec/libsass-closed-issues/issue_759.hrx]: Now there is! https://github.com/sass/sass/issues/2607

**Pull request merged**